Source/Core/AST/AbsyCmd.cs

```diff
@@ -3846,6 +3848,11 @@ protected override Cmd ComputeDesugaring(PrintOptions options)
               a.Attributes = attrCopy;
             }
 
+            // Do this after copying the attributes so it doesn't get overwritten
```

Comment:
Created #743 to track this, as I think we need to do it very carefully.

Reply:
> I think we need to do it very carefully

Why? Splitting a method up into pieces can be done in a mostly automated way using Rider. Same for moving a class into a separate file.

Reply:
Yeah, but reorganizing those methods so that they are well-organized and use better abstractions, rather than just slicing them into pieces, would require more thought.

Reply:
I'm paraphrasing an offline discussion I had with @atomb on the above, for the sake of getting some additional eyes on it.

@atomb prefers keeping the pacify implementations for various commands in a single method, since that allows having them be vertically adjacent to each other which follows the structure of Figure 3 in [this paper](https://users.soe.ucsc.edu/~cormac/papers/popl01.pdf), which provides the best way of doing a side-by-side comparison of the two.

I would prefer keeping the code for different commands separate. I would even prefer having an abstract `Pacify` method the `Cmd` class. The number of commands is not fixed, we might add more in the future, and I think in this way the codebase scales well to that.

I believe that while the paper is structured by putting the concepts at the top-level, and then discussing all commands within each concept, this doesn't mean that's also the right structure for the code. It's easier to discuss just one pair of concept and command at a time, and while the paper doesn't have room to go that slow, in our codebase we can use as many separators, such as files and methods, as we like. Also, the code is more verbose so there is more need to keep the scope of each block to a minimum.

I don't think optimizing for comparability between the code and the paper is important. Modern IDEs make it easy to navigate between the code for various commands, so I don't think there's a particular difficulty in making the comparison in any case.